src/TimeSeries/index.js

```diff
@@ -0,0 +1,743 @@
+import * as tf from "@tensorflow/tfjs";
+import callCallback from "../utils/callcallback";
+import handleArguments from "../utils/handleArguments";
+import NeuralNetwork from "./timeSeries";
+import NeuralNetworkData from "./timeSeriesData";
+import nnUtils from "../NeuralNetwork/NeuralNetworkUtils";
+import NeuralNetworkVis from "../NeuralNetwork/NeuralNetworkVis";
+
+import tsUtils from "./timeSeriesUtils";
+
+const DEFAULTS = {
+  inputs: [],
+  outputs: [],
+  dataUrl: null,
+  modelUrl: null,
+  layers: [],
+  task: null,
+  dataModality: null,
+  debug: false,
+  learningRate: 0.2,
+  hiddenUnits: 16,
+};
+/*
+as far as the p5 sketch is concerned, it will directly call only a few functions in the class,
+these are the following:
+
+model.addData - Done
+model.saveData, model etc
+model.train
+model.classify/predict etc
+
+
+No image classification
+No neural evolution
+*/
+
+class timeSeries {
+  constructor(options, callback) {
+    this.options =
+      {
+        ...DEFAULTS,
+        ...options,
+      } || DEFAULTS;
+
+    this.neuralNetwork = new NeuralNetwork();
+    this.neuralNetworkData = new NeuralNetworkData();
+    this.neuralNetworkVis = new NeuralNetworkVis();
+
+    this.data = {
+      training: [],
+    };
+
+    this.init = this.init.bind(this);
+
+    this.ready = callCallback(this.init(), callback);
+  }
+
+  async init() {
+    if (this.options.dataUrl) {
+      await this.loadDataFromUrl(this.options.dataUrl);
+    } else if (this.options.modelUrl) {
+      await this.load(this.options.modelUrl);
+    }
+    return this;
+  }
+
+
+  /**
+   * ////////////////////////////////////////////////////////////
+   *                   Add and Format Data
+   * ////////////////////////////////////////////////////////////
+   */
+
+  /* adding data: can only accept the following formats:
+     - for xInputs:
+      1. Sequence of objects (array of objects) 
+        [{x: , y: },{x: , y: },{x: , y: },{x: , y: }]
+      2. Sequence of arrays (array of array, order matters)
+        [[],[],[],[]]
+      3. Sequence of values (inputlabels should be provided by user)
+        [[,,,,,]] e.g. shape = {inputLabels: ['x','y']} will become [{x: , y: },{x: , y: },{x: , y: },{x: , y: }]
+  
+    - for yInputs:
+      1. similar to neural network, so use same logic
+
+    - at the end of the adding data, the data is formatted to a sequence of objects similar to 1 of xinputs
+ 
+    - new parameter dataModality, either spatial or sequential, spatial uses cnn1d and sequential uses lstm
```

Comment:
Hi professor, I agree with this comment, I think that the dataModality is a little bit confusing, especially when the inputs are strings! as much as I want the data to be automatically categorized, unfortunately I currently cannot see a practical way to know, as other data such as weather also uses numbers exclusively, so instead of this, I changed data modality into `spatialData` which accepts a boolean, by default without setting it, it is using lstm which works for all data sequences so its not a problem, so if users want more accuracy for less datasets in coordinate data they can just set this as true, I think this is the most pragmatic way for now.

Reply:
Still thinking that having different `tasks` might be more clear and educational compared to a flag (where we don't quite know what its doing without looking at the source code).

Reply:
Hi Gottfried, I reverted to the old way without the flag. I implemented a change for this and changed the `dataModality` into `dataMode` to make it shorter. I keep the `spatial` but changed the `sequential` to linear, tell me what you think about these changes